src/BaGet.Core/Search/DatabaseSearchService.cs

```diff
@@ -106,15 +106,35 @@ public DatabaseSearchService(IContext context, IFrameworkCompatibilityService fr
             };
         }
 
-        public Task<AutocompleteResponse> ListPackageVersionsAssync(
-            string packageId,
-            bool includePrerelease,
-            bool includeSemVer2,
+        public async Task<AutocompleteResponse> ListPackageVersionsAsync(
+            VersionsRequest request,
             CancellationToken cancellationToken)
         {
-            // TODO: Support versions autocomplete.
-            // See: https://github.com/loic-sharma/BaGet/issues/291
-            throw new NotImplementedException();
+            IQueryable<Package> search = _context.Packages;
+
+            if (!string.IsNullOrEmpty(request.Query))
+            {
+                var query = request.Query.ToLower();
+                search = search.Where(p => p.Id.ToLower().Equals(query));
+            }
+
+            search = AddSearchFilters(
+                search,
+                request.IncludePrerelease,
+                request.IncludeSemVer2,
+                request.PackageType,
+                frameworks: null);
+
+            var results = await search
+                .Select(p => p.Version.ToString())
```

Comment:
You can use [`NormalizedVersionString`](https://github.com/loic-sharma/BaGet/blob/f0ee2c57c221fd6f0e866f051eb5892e464b85f0/src/BaGet.Core/Entities/Package.cs#L67) directly instead (the `Version` property has to parse the version string)

```suggestion
                .Select(p => p.NormalizedVersionString)
```